windows/src/engine/insthelper/Keyman.System.Install.EnginePostInstall.pas

```diff
@@ -25,6 +25,39 @@ function ReportFailure(hInstall: MSIHANDLE; const func: string; code: UINT): UIN
   Result := code;
 end;
 
+
+function UpdateState: Boolean;
+var
+  UpdateStr : UnicodeString;
+  UpdatePBytes : PByte;
+  hk: Winapi.Windows.HKEY;
+  updateData: Cardinal;
+begin
+
+  Result := False;
+  UpdateStr := 'usPostInstall';
+  if RegOpenKeyEx(HKEY_LOCAL_MACHINE, PChar(SRegKey_KeymanEngine_CU), 0, KEY_ALL_ACCESS, hk) = ERROR_SUCCESS then
```

Comment:
Yes this is should be CU, this PR is missing that commit. I moved it to CU otherwise you were blocked from updating the state if not in elevated. 

In the Next PR this is moved from this dll and updated as part of firstrun in kmshell.

>Are you intending to use HKCU? Who owns the update state -- machine or current user? Who is the current user here (think about admin installations)?

My testing has only been users who can elevate to admin permissions. For a user that does not have admin and has to use another user to elevate ... this could be a problem. It may have to go back to a seralized state on the disk. hmmm